src/network/modals/CreateSponsorshipModal.tsx

```diff
@@ -107,6 +111,23 @@ export default function CreateSponsorshipModal({
     const invalidMinStakeDuration =
         !!initialAmount && !!payoutRate && extensionInDays < minStakeDuration
 
+    const [maxPenaltyPeriod, setMaxPenaltyPeriod] = useState<number>()
+
+    const tooLongMinStakeDuration =
+        !!maxPenaltyPeriod && minStakeDuration > maxPenaltyPeriod
+
+    useEffect(() => {
+        getConfigFromChain().then((config) => {
+            if (isMounted() && config) {
```

Comment:
Let's avoid `useIsMounted` hook, please. Doesn't hurt here but yeah, just saying.

Reply:
Why? It helps to handle those situations where you'd try to update a component that was unmounted.